### PR TITLE
chore(samples): create sample apps from manifests

### PR DESCRIPTION
**Pull request by tthvo**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits): `git commit -S -m "YOUR_COMMIT_MESSAGE"`
_______________________________________________

Fixes: #454 

## Description of the change:

- Create sample applications from manifests so that we set env and security context easily.

## Motivation for the change:

See #454

## How to manually test:

`make sample_app`

### CODE REVIEW

**Review by andrewazores — APPROVED** [review]

**Comment by andrewazores** [comment]
New CI looks like it's progressing well.

**Comment by andrewazores** [comment]
I spoke too soon:

image

https://github.com/cryostatio/cryostat-operator/actions/runs/5084122461/jobs/9136007090?pr=590

**Comment by andrewazores** [comment]
Same thing on the merge to `main` actually:

https://github.com/cryostatio/cryostat-operator/actions/runs/5084115155/jobs/9135990797

#588 #589

**Comment by ebaron** [comment]
The GHCR secret wasn't available to this repo. Added it now.

**Comment by tthvo** [comment]
Nice CI is working now ^^

**Review by andrewazores — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -430,21 +430,17 @@ scorecard-build: custom-scorecard-tests
...
-.PHONY: sample_app_quarkus
-sample_app_quarkus: undeploy_sample_app_quarkus
-	$(call new-sample-app,quay.io/andrewazores/quarkus-test:0.0.10)
-	$(CLUSTER_CLIENT) patch svc/quarkus-test -p '{"spec":{"$setElementOrder/ports":[{"port":9097},{"port":8080}],"ports":[{"name":"jfr-jmx","port":9097}]}}'
-
-.PHONY: undeploy_sample_app_quarkus
-undeploy_sample_app_quarkus:
-	- $(CLUSTER_CLIENT) delete all -l app=quarkus-test
+undeploy_sample_app:
+	$(CLUSTER_CLIENT) delete --ignore-not-found=$(ignore-not-found) -f config/samples/sample-app.yaml
```
Comment (ebaron) [inline Makefile:443]: I think we need `$(SAMPLE_APP_FLAGS)` here as well to make sure we delete from the right namespace.

File: Makefile [hunk 2]
```diff
@@ -456,14 +452,9 @@ sample_app_agent: undeploy_sample_app_agent
...
 
 .PHONY: undeploy_sample_app_agent
 undeploy_sample_app_agent:
-	- $(CLUSTER_CLIENT) delete -f config/samples/sample-app-agent.yml
-
-define new-sample-app
-@if [ ! "$(CLUSTER_CLIENT)" = "oc" ]; then echo "CLUSTER_CLIENT must be 'oc' for sample app deployments" && exit 1; fi
-$(CLUSTER_CLIENT) new-app $(1)
-endef
+	- $(CLUSTER_CLIENT) delete --ignore-not-found=$(ignore-not-found) -f config/samples/sample-app-agent.yaml
```
Comment (ebaron) [inline Makefile:460]: Same here

File: Makefile (hunk 1 above)
Comment (tthvo) [reply Makefile]: oh yes! Missed that! Thanks!

**Review by ebaron — APPROVED** [review]

**Pull request merged**